src/Raven.Server/RavenServer.cs

```diff
@@ -513,14 +513,37 @@ private void StartMonitoringCpuCredits()
             var sw = Stopwatch.StartNew();
             Stopwatch err = null;
 
+            int currentRetry = 0;
             try
             {
-                UpdateCpuCreditsFromExec();
+                // This is the first sync after a restart, so we want to retry until we get a valid result. Next sync is in 30 minutes.
+                while (true)
+                {
+                    try
+                    {
+                        UpdateCpuCreditsFromExec();
+                        break;
+                    }
+                    catch (Exception ex)
+                    {
+                        currentRetry++;
+
+                        // If we weren't able to sync at startup, we'll assume we have enough credits and let RavenDB perform in the meantime.
+                        // We don't want RavenDB to (false) throttle unless it knows for sure that it doesn't have credits.
+                        CpuCreditsBalance.RemainingCpuCredits = CpuCreditsBalance.MaxCredits;
+                        
+                        if (currentRetry > 5)
+                        {
+                            throw;
+                        }
+                    }
+                    Task.Delay(TimeSpan.FromMinutes(1));
```

Comment:
Why do we retry at this length?

Reply:
I figured we don't get a result because of the cloud provider. We need to wait but for how long? Well, I just thought that one minute is a long enough period to get new results.

Reply:
The question is, why do we need to wait at all? We already have a periodic check, no?
Why hold up the thread for so long?

Reply:
I merged the logic with the main loop, it will try to sync every minute until we get a valid result.